frontend/src/app/organization/organization-create/organization-create.component.ts

```diff
@@ -0,0 +1,80 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { switchMap, tap } from 'rxjs/operators';
+import { Subscription } from 'rxjs/Subscription';
+
+import { AuthService } from '../../auth.service';
+import { Organization } from '../organization.model';
+import { OrganizationService } from '../organization.service';
+
+@Component({
+  selector: 'volontulo-organization-create',
+  templateUrl: './organization-create.component.html',
+  styleUrls: ['./organization-create.component.scss']
+})
+export class OrganizationCreateComponent implements OnInit, OnDestroy {
+  createForm: FormGroup;
+  id: number;
+  inEditMode = false;
+  message = '';
+  statusSubscription: Subscription;
+
+  constructor(
+    private authService: AuthService,
+    private activatedRoute: ActivatedRoute,
+    private fb: FormBuilder,
+    private organizationService: OrganizationService,
+    private router: Router,
+    ) {}
+
+  ngOnInit() {
+    this.createForm = this.fb.group({
+       'name': this.fb.control(null, Validators.required),
+       'address': this.fb.control(null, Validators.required),
+       'description': this.fb.control(null, Validators.required),
+      });
+
+    this.activatedRoute.params
+      .pipe(
+        tap(params => {
+          if (params.organizationId) {
+            this.inEditMode = true;
+            this.id = params.organizationId;
+            this.organizationService.getOrganization(params.organizationId);
+          }
+        }),
+        switchMap(() => this.organizationService.organization$)
+      )
+      .subscribe(
+        (organization: Organization) => this.createForm.patchValue(organization)
+      );
+
+    this.statusSubscription = this.organizationService.createStatus$
+      .subscribe(
+        (response) => {
+          if (response.status === 'success') {
+            this.authService.getCurrentUser();
```

Comment:
As we spoke privately - I think it would be better to update user manually from angular app then sending a request to server. Think of adding a method to authService:
```
setCurrentUser(user: User) {
  this.changeUserEvent.next(user);
}
```
or something like that.